devtools/etdump/etdump_filter.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <executorch/devtools/etdump/etdump_filter.h>
+
+#include <executorch/runtime/core/error.h>
+
+
+using ::executorch::runtime::DelegateDebugIntId;
+using ::executorch::runtime::Error;
+using ::executorch::runtime::kUnsetDelegateDebugIntId;
+
+namespace executorch {
+namespace etdump {
+
+ETDumpFilter::ETDumpFilter()
+    : regex_count_(0), range_start_(0), range_end_(0) {}
+
+Result<bool> ETDumpFilter::add_regex(const char* pattern) {
+  if (regex_count_ >= MAX_REGEX_PATTERNS) {
+    return Error::OutOfResources; // Error code for exceeding max patterns
+  }
+  size_t len = strlen(pattern);
+  if (len >= MAX_PATTERN_LENGTH) {
+    return Error::InvalidArgument; // Pattern too long
+  }
+  strcpy(regex_patterns_[regex_count_], pattern);
```

Comment:
nit: you already have the length, so use it -- `std::memcpy(regex_patterns_[regex_count_], pattern, len)`